frontend/app/src/components/cloud/billing/subscription.tsx

```diff
@@ -51,6 +53,10 @@ export const Subscription: React.FC<SubscriptionProps> = ({
   const { handleApiError } = useApiError({});
   const [portalLoading, setPortalLoading] = useState(false);
 
+  useEffect(() => {
+    setShowAnnual(active?.period?.includes('yearly') || false);
+  }, [active]);
```

Comment:
I know this was just moved up from below, but I'd think we could get rid of this effect hook and use derived state?

```typescript
const showAnnual = active?.period?.includes("yearly") || false
```